src/preferences/dialog/dlgpreflibrary.cpp

```diff
@@ -174,14 +171,15 @@ void DlgPrefLibrary::slotUpdate() {
     case ADD_TO_AUTODJ_TOP:
             radioButton_dbclick_top->setChecked(true);
             break;
-    case EDIT_METADATA:
-            radioButton_dbclick_edit_metadata->setChecked(true);
-            break;
     default:
             radioButton_dbclick_deck->setChecked(true);
             break;
     }
 
+    checkBoxEditMetadata->setChecked(
+            m_pConfig->getValue(
+                        ConfigKey("[Library]","EditMetadata"), true));
```

Comment:
IMO this is much too dangerous to be enabled by default.

Reply:
Since this was the default in 2.0 I like to keep it. Windows and Mac user are used to that "feature"

Reply:
I don't think either of those are good reasons to make this on by default. This behavior is so unexpected and confusing that no one could even describe quite what the problem was until I started working on the code and reading the Qt documentation. If users have to read the "Edit metadata after clicking selected track" description in the preferences before enabling this behavior, I think they will have an easier time understanding how it works and therefore understand how to use it intentionally and avoid activating it unintentionally.

Reply:
For feeling is that this topic has stirred up, because of a single users bug report. 
All Windows an Mac users expect the feature and can handle it in their file manager. 
This feature was enabled for ages in Mixxx, without anyone else complaining.
But we had complains during the short phase with the double click edit, loosing this feature. 

So I assume that many user would feel the disabled feature as a regression. It might took a while to discover the check-box to re-enable it.

I do not use this feature and I also have never accidentally enabled it. So I do not care personally about the default. 

How do others think?

Reply:
@daschuer are you referring to https://bugs.launchpad.net/mixxx/+bug/1665583 ? If so, let me reassure you, this has been giving mac users a hard time using mixxx to the point where live usage becomes questionable due to unusable waveforms when accidentally getting into track edit mode. The bug was files feb 2017 and only disabling edit mode / adding an option in prefs makes plex usable for macOS users on a laptop with trackpad. Let me know if I am missing something. My experience is limited to macOS, so this may be working great on linux / win.

Reply:
For me this never was an issue, maybe because I'm on Linux and because I don't have to use keyboard shortcuts. Direct editing is a great feature, I use it a lot for writing track comments.

Let's say I run into the issue that track editing interferes with my track load workflow and annoys me, I'd definitely dig through the Preferences options and see if I can disable it. This is definitely biased since I know how much Mixxx can be customized.

Or put the other way around (off by default): I wouldn't expect that feature and probably wouldn't miss it, so I wouldn't go looking for it.

So I vote for default=ON

Reply:
>  All Windows an Mac users expect the feature and can handle it in their file manager.

I've never used Mac very much, but long ago when I used to use Windows I do not remember this click selected item to edit behavior. I remember pressing F2 to rename files. Also, the use cases of file managers and a DJ program are quite different. File managers are not used for live performance.

Reply:
I was reffering https://bugs.launchpad.net/mixxx/+bug/1743238

Reply:
Note that https://bugs.launchpad.net/mixxx/+bug/1743238 has 4 people marked as being affected. It is not just one user opening a bug report.

Reply:
@daschuer In that case please have a look at the bug linked under "2." which elaborates on a nasty side effects of edit mode on macOS. I do not want to over-complicate this discussion. Having an option to disable editing is just fine for me, thus I do not have strong feelings about the default.

I currently see two concerns regarding defaulting to on:
1. writing to files may lead to unwanted results, i.e. having files named "e" "ff" or whatever user types when being in edit mode accidentally. I can confirm that 100% of the time I went to edit mode it was by accident. (could be solved by having write to files disabled by default and show an informative warning when first enabled - both is the case currently iirc?)
2. macOS side effect https://bugs.launchpad.net/mixxx/+bug/1665583

Reply:
As if on cue, we have [another macOS user reporting the same bug](https://bugs.launchpad.net/mixxx/+bug/1759433).

Reply:
I'm confused, the bug was that "it's too easy to trigger metadata editing" but this user is reporting "If I click any field in the library (say, to change a song's name/artist), the UI goes down to 2 FPS."  How is that the same bug?

Reply:
> macOS side effect

This sounds like a separate issue that needs to be addressed on that platform.  Disabling click-to-edit would just be sweeping the problem under the carpet.

Reply:
> How do others think?

We've had click-to-edit for many years without complaints.  I think the problem is important enough to create the preference that we have done here, but I feel comfortable defaulting to the existing behavior.  We can update the bug and point people to the new preference.

It's very hard to get a sense of what percentage of users a particular mis-feature affects, and mixxx has a habit of implementing features for single-digit user counts vs the ~million downloads we get per year.  Sure, not everyone who experiences a problem will register for a bug account and find the same bug, so at some point we need to use our own judgement.

Reply:
> We've had click-to-edit for many years without complaints.

It is a little frustrating to have to repeat myself:
> Note that https://bugs.launchpad.net/mixxx/+bug/1743238 has 4 people marked as being affected. It is not just one user opening a bug report.

Reply:
> This sounds like a separate issue that needs to be addressed on that platform. Disabling click-to-edit would just be sweeping the problem under the carpet.

Well we have no macOS developers and we are try to get a release out imminently. This is the time for quick hacks, not debugging complicated issues.

Reply:
I'm aware that the bug exists, but it was reported in January of this year and we've had this feature for many years, so this seems to be a new problem on OSX only.  I suspect something changed on that platform, like the addition of "force clicks" which is VERY confusing for users who don't know what it is (including myself).  Maybe "force click" is enabling edit mode?

What about making the default off for OSX only?

Reply:
The UI lagging with editing metadata from the track table is only on macOS, but the high risk of accidentally editing metadata is for all OSs. This has been a problem for me on GNU/Linux with Mixxx for as long as I've been using it. I didn't report a bug for it earlier because the behavior is so unintuitive that I couldn't figure out exactly what triggered the accidental edits.

Having different defaults for different platforms doesn't seem like a great idea to me.

Reply:
Fair enough.  My preference is to leave it on by default but as long as there's adequate documentation I'm ok with the default being either way

Reply:
If the default is changed to off, I think the new inverted "Prevent editing of metadata by clicking selected track" wording is odd.